File: 프로그래머스/0/181883. 수열과 구간 쿼리 1/수열과 구간 쿼리 1.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries) {
    vector<int> answer = arr;
    for(int i = 0; i < queries.size(); i++)
    {
        int s = queries[i][0];
        int e = queries[i][1];
        for(int j = s; j <= e; j++)
            answer[j] += 1;
    }
    return answer;
}

File: 프로그래머스/0/181883. 수열과 구간 쿼리 1/수열과 구간 쿼리 1.cs
using System;

public class Solution {
    public int[] solution(int[] arr, int[,] queries) {
        int[] answer = arr;
        for(int i = 0 ; i < queries.GetLength(0);i++)
        {
            int s = queries[i,0];
            int e = queries[i,1];
            for(int j = s ;j <= e;j++)
            {
                if(0 <= j && j <= arr.Length)
                {
                    arr[j] += 1;
                }
            }
        }
        return answer;
    }
}